File: backend/src/main/java/com/happy/friendogly/footprint/dto/response/detail/PetDetail.java
package com.happy.friendogly.footprint.dto.response.detail;

import com.happy.friendogly.pet.domain.Gender;
import com.happy.friendogly.pet.domain.Pet;
import com.happy.friendogly.pet.domain.SizeType;
import java.time.LocalDate;

public record PetDetail(
        String name,
        String description,
        LocalDate birthDate,
        SizeType sizeType,
        Gender gender,
        String imageUrl
) {
    public PetDetail(Pet pet) {
        this(
                pet.getName().getValue(),
                pet.getDescription().getValue(),
                pet.getBirthDate().getValue(),
                pet.getSizeType(),
                pet.getGender(),
                pet.getImageUrl()
        );
    }
}

File: android/app/src/main/java/com/happy/friendogly/presentation/ui/playground/model/PetDetail.kt
package com.happy.friendogly.presentation.ui.playground.model

import com.happy.friendogly.domain.model.Gender
import com.happy.friendogly.domain.model.SizeType
import kotlinx.datetime.LocalDate

data class PetDetail(
    val name: String,
    val description: String,
    val birthDate: LocalDate,
    val sizeType: SizeType,
    val gender: Gender,
    val imageUrl: String,
)
